### PR TITLE
Allow the main game loop to scale the delta time

### PR DESCRIPTION
**Pull request by ZehMatt**
Which results in this:

https://www.youtube.com/watch?v=dNNmagd1nOs

The final goal is to get rid of the inner loop that drives the game speed and have input/windows update at fixed time, I already have scrolling fixed locally but I want to do smaller changes over time.

### CODE REVIEW

**Comment by janisozaur** [comment]
Oh my, floats in the core, no like.

**Comment by ZehMatt** [comment]
> Oh my, floats in the core, no like.

It just accumulates time in the one float and tweens with a float. Nothing of the game will change by this.

**Comment by janisozaur** [comment]
ah, carry on then.

File: src/openrct2/Context.cpp [hunk 1]
```diff
@@ -1220,6 +1221,16 @@ namespace OpenRCT2
         {
             return &_newVersionInfo;
         }
+
+        void SetTimeScale(float newScale) override
+        {
+            _timeScale = std::clamp(newScale, 0.0f, 5.0f);
```
Comment (janisozaur) [inline src/openrct2/Context.cpp:1227]: Seems the two values are related?
```suggestion
            _timeScale = std::clamp(newScale, 0.0f, GAME_MAX_UPDATES);
```
Also does it make sense to allow 0 scale?

Comment (ZehMatt) [reply src/openrct2/Context.cpp:1227]: Good point, I think 0.1 should be small enough.

Comment (ZehMatt) [reply src/openrct2/Context.cpp:1227]: GAME_MAX_UPDATES is used for something else, I accidentally changed that.

**Review by tupaschoal — APPROVED** [review]
LGTM, though I wonder what the motivation was (testing?)

**Comment by ZehMatt** [comment]
There are a couple of things why I did this.
1. Testing the tweening when opening/closing rides in slow motion.
2. Preparing for future code clean to get rid of the nested loop to speed up the game.
3. Its cool to control the time scale as I wish.

**Pull request merged**